modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,522 @@
+	public String getCssClass() {
+		if ((super.getCssClass() == null) && (_userCard != null)) {
+			if (_userCard.getCssClass() != null) {
+				return _userCard.getCssClass();
+			}
+
+			if (_userCard.getElementClasses() != null) {
+				return _userCard.getElementClasses();
+			}
+		}
+
+		return super.getCssClass();
+	}
+
+	public String getHref() {
+		if ((_href == null) && (_userCard != null)) {
+			return _userCard.getHref();
+		}
+
+		return _href;
+	}
+
+	public String getIcon() {
+		if ((_icon == null) && (_userCard != null)) {
+			return _userCard.getIcon();
+		}
+
+		return _icon;
+	}
+
+	@Override
+	public String getId() {
+		if ((super.getId() == null) && (_userCard != null)) {
+			return _userCard.getId();
+		}
+
+		return super.getId();
+	}
+
+	public String getImageAlt() {
+		if ((_imageAlt == null) && (_userCard != null)) {
+			return _userCard.getImageAlt();
+		}
+
+		return _imageAlt;
+	}
+
+	public String getImageSrc() {
+		if ((_imageSrc == null) && (_userCard != null)) {
+			return _userCard.getImageSrc();
+		}
+
+		return _imageSrc;
+	}
+
+	public String getInputName() {
+		if ((_inputName == null) && (_userCard != null)) {
+			return _userCard.getInputName();
+		}
+
+		return _inputName;
+	}
+
+	public String getInputValue() {
+		if ((_inputValue == null) && (_userCard != null)) {
+			return _userCard.getInputValue();
+		}
+
+		return _inputValue;
+	}
+
+	public List<LabelItem> getLabels() {
+		if ((_labels == null) && (_userCard != null)) {
+			return _userCard.getLabels();
+		}
+
+		return _labels;
+	}
+
+	public String getName() {
+		if ((_name == null) && (_userCard != null)) {
+			return _userCard.getName();
+		}
+
+		return _name;
+	}
+
+	public String getSubtitle() {
+		if ((_subtitle == null) && (_userCard != null)) {
+			return _userCard.getSubtitle();
+		}
+
+		return _subtitle;
+	}
+
+	public UserCard getUserCard() {
+		return _userCard;
+	}
+
+	public String getUserColorClass() {
+		if ((_userColorClass == null) && (_userCard != null)) {
+			return _userCard.getUserColorClass();
+		}
+
+		return _userColorClass;
+	}
+
+	public Boolean isDisabled() {
+		if (_disabled == null) {
+			if (_userCard != null) {
+				return _userCard.isDisabled();
+			}
+
+			return false;
+		}
+
+		return _disabled;
+	}
+
+	public Boolean isSelectable() {
+		if (_selectable == null) {
+			if (_userCard != null) {
+				return _userCard.isSelectable();
+			}
+
+			return true;
+		}
+
+		return _selectable;
+	}
+
+	public Boolean isSelected() {
+		if (_selected == null) {
+			if (_userCard != null) {
+				return _userCard.isSelected();
+			}
+
+			return false;
+		}
+
+		return _selected;
+	}
+
+	public void setActionDropdownItems(List<DropdownItem> actionDropdownItems) {
+		_actionDropdownItems = actionDropdownItems;
+	}
+
+	public void setDisabled(Boolean disabled) {
+		_disabled = disabled;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	public void setGroupName(String groupName) {
```

Comment:
Since this is a new class where we don't use the getter we don't need to add it. We need to add the setter since that attribute it's present in the `tld`, but marking it as deprecated.

Reply:
Why add setter then? It is not used either. We could just deprecate attribute in `tld`. Is the idea to fix bug from current tag, asserting that setter must be there if attribute is defined in `tld`?

Reply:
If someone is using a deprecated attribute without a defined setter it will crash on runtime. Since we can't remove the attribute from the `tld` until the next version I think it's safer (and easier) to just add the `setter` marked as deprecated and remove it in the next version together with the attribute in the tld.

Reply:
OK, makes sense :+1: